### PR TITLE
Feature/no dev build

### PR DESCRIPTION
**Pull request by Caleb-Cohen**
# Description

This pull request is designed to allow development to not need the build folder to work. Discord authentication will now function on the front end of port 3000. This will also detect what what redirect should be used depending on if application is in development or production. 

**A new .ENV variable should be added**
`BASE_URL = http://localhost:3000/`

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Issue
- [x] Is this related to a specific issue? Is so please specify: #122 

# Checklist:

- [x] I have executed `npm run lint` and resolved any outstanding errors. Most issues can be solved by executing `npm run format`
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "start": "cd server && node server.js",
     "dev": "npm --prefix server run dev",
-    "build": "npm --prefix server run build:ui",
+    "build": "npm --prefix client run build",
```
Comment (Caleb-Cohen) [inline package.json:9]: @JoeSangine can you explain this change? I think this may cause issues with production.

File: server/package.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "scripts": {
     "start": "node server.js",
-    "dev": "nodemon server.js",
-    "build:ui": "rm -rf build && cd ../client && npm run build --prod && cp -r build ../server"
```
Comment (Caleb-Cohen) [inline server/package.json:5]: @JoeSangine Same question as my previous comment. Why is build commands being taken out to the server/package.json?

**Review by Caleb-Cohen — COMMENTED** [review]
Added some comments to @JoeSangine regarding build command changes. I'm pretty confident they will break production, and it's unclear to me why the changes were made.

File: package.json (hunk 1 above)
Comment (JoeSangine) [reply package.json:9]: So the backend doesn't actually have the code to build, the backend was telling the frontend to build - but was being told by the root to build. So insted of root -> backend -> frontend to build, we say root -> frontend to build

File: server/package.json (hunk 2 above)
Comment (JoeSangine) [reply server/package.json:5]: see comment above

File: client/package.json [hunk 3]
```diff
@@ -6,5 +6,8 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject",
     "dev": "json-server --watch src/test/events.json --port 2121"
+  },
+  "dependencies": {
+    "http-proxy-middleware": "^2.0.6"
   }
```
Comment (JoeDravarol) [inline client/package.json:12]: No package/depencies should be installed inside of client folder. This specific package should be installed under dev dependencies in the root folder.

File: server/routes/main.js [hunk 4]
```diff
@@ -3,14 +3,14 @@ const router = express.Router();
 const passport = require("passport");
 
 //Main Routes - simplified for now
-
+const baseUrl = process.env.BASE_URL || 'http://localhost:3000'
```
Comment (JoeDravarol) [inline server/routes/main.js:6]: This and the env variable should be changed to 0AUTH_REDIRECT_URL for better clarity.

File: server/server.js [hunk 5]
```diff
@@ -8,19 +8,15 @@ const MongoStore = require("connect-mongo")(session);
...
 const logger = require("morgan");
+//Use .env file in config folder
+require("dotenv").config({ path: "./config/.env" });
 const connectDB = require("./config/database");
 const mainRoutes = require("./routes/main");
 const eventsRoutes = require("./routes/events");
 
-//Use .env file in config folder
-require("dotenv").config({ path: "./config/.env" });
-
```
Comment (JoeDravarol) [inline server/server.js:17]: Can you explain why this was move upward? Were issue with files not being able to access env variables?

File: server/server.js [hunk 6]
```diff
@@ -42,7 +38,7 @@ app.use(
 );
 
 // Render React as View
-app.use(express.static(path.join(__dirname, "build")));
+app.use(express.static(path.join(__dirname, "..", "client", "build")));
```
Comment (JoeDravarol) [inline server/server.js:41]: I don't think we should access the build folder this way but that's my personal opinion.

File: server/server.js [hunk 7]
```diff
@@ -54,6 +50,9 @@ app.use(flash());
 //Setup Routes For Which The Server Is Listening
 app.use("/", mainRoutes);
 app.use("/events", eventsRoutes);
+app.get("'", (req, res) => {
+  res.sendFile(path.join(__dirname, "..", "client", "build", "index.html"));
+})
```
Comment (JoeDravarol) [inline server/server.js:55]: Same comment as above. (Accessing build folder)

**Review by JoeDravarol — CHANGES_REQUESTED** [review]

**Pull request merged**